src/Reflection/InitializerExprTypeResolver.php

```diff
@@ -1616,63 +1626,93 @@ private function integerRangeMath(Type $range, BinaryOp $node, Type $operand): T
-				$max = $rangeMax !== null && $operand->getMax() !== null ? $rangeMax * $operand->getMax() : null;
+			if ($min === -INF) {
+				$min = null;
 			}
-
-			if ($min !== null && $max !== null && $min > $max) {
-				[$min, $max] = [$max, $min];
-			}
-
-			// invert maximas on multiplication with negative constants
-			if ((($range instanceof ConstantIntegerType && $range->getValue() < 0)
-					|| ($operand instanceof ConstantIntegerType && $operand->getValue() < 0))
-				&& ($min === null || $max === null)) {
-				[$min, $max] = [$max, $min];
+			if ($max === INF) {
+				$max = null;
 			}
-
 		} else {
 			if ($operand instanceof ConstantIntegerType) {
 				$min = $rangeMin !== null && $operand->getValue() !== 0 ? $rangeMin / $operand->getValue() : null;
 				$max = $rangeMax !== null && $operand->getValue() !== 0 ? $rangeMax / $operand->getValue() : null;
 			} else {
-				$min = $rangeMin !== null && $operand->getMin() !== null && $operand->getMin() !== 0 ? $rangeMin / $operand->getMin() : null;
-				$max = $rangeMax !== null && $operand->getMax() !== null && $operand->getMax() !== 0 ? $rangeMax / $operand->getMax() : null;
-			}
+				// Avoid division by zero when looking for the min and the max by using the closest int
+				$operandMin = $operandMin !== 0 ? $operandMin : 1;
+				$operandMax = $operandMax !== 0 ? $operandMax : -1;
+
+				if (
+					($operandMin < 0 || $operandMin === null)
+					&& ($operandMax > 0 || $operandMax === null)
+				) {
+					$result = TypeCombinator::union(
+						$this->integerRangeMath($range, $node, IntegerRangeType::fromInterval($operandMin, 0)),
+						$this->integerRangeMath($range, $node, IntegerRangeType::fromInterval(0, $operandMax))
+					)->toNumber();
+
+					if ($result->equals(new UnionType([new IntegerType(), new FloatType()]))) {
+						return new BenevolentUnionType([new IntegerType(), new FloatType()]);
+					} else {
+						return $result;
+					}
+				}
+				if (
+					($rangeMin < 0 || $rangeMin === null)
+					&& ($rangeMax > 0 || $rangeMax === null)
+				) {
+					$result = TypeCombinator::union(
+						$this->integerRangeMath(IntegerRangeType::fromInterval($rangeMin, 0), $node, $operand),
+						$this->integerRangeMath(IntegerRangeType::fromInterval(0, $rangeMax), $node, $operand)
+					)->toNumber();
+
+					if ($result->equals(new UnionType([new IntegerType(), new FloatType()]))) {
+						return new BenevolentUnionType([new IntegerType(), new FloatType()]);
+					} else {
+						return $result;
+					}
+				}
+
+				$rangeMinSign = ($rangeMin ?? -INF) <=> 0;
+				$rangeMaxSign = ($rangeMax ?? INF) <=> 0;
+
+				$min1 = $operandMin !== null ? ($rangeMin ?? -INF) / $operandMin : $rangeMinSign * -0.1;
+				$min2 = $operandMax !== null ? ($rangeMin ?? -INF) / $operandMax : $rangeMinSign * 0.1;
+				$max1 = $operandMin !== null ? ($rangeMax ?? INF) / $operandMin : $rangeMaxSign * -0.1;
+				$max2 = $operandMax !== null ? ($rangeMax ?? INF) / $operandMax : $rangeMaxSign * 0.1;
```

Comment:
I basically use the same tips than for multiplication, replacing `null` by `INF`. But a division by `INF` is not great, giving the float 0 or NAN.

I can still simulate this division by saying the min/max is really near from 0 with the correct sign.